internal/graph/graph.go

```diff
@@ -143,6 +148,24 @@ func (g *Graph) InsertConstructor(ctor interface{}) error {
 	return nil
 }
 
+// ValidateReturnTypes validates if ctor's return type is already insterted in the graph
+func (g *Graph) ValidateReturnTypes(ctype reflect.Type) error {
+	g.RLock()
+	defer g.RUnlock()
+	objMap := make(map[reflect.Type]bool)
```

Comment:
Its a throwaway map for start up. I am pretty sure we can live with a boolean rather than handling empty structs.:)